java/src/com/github/bazel_contrib/contrib_rules_jvm/junit5/README.md

```diff
@@ -0,0 +1,280 @@
+
+#### Suite
+
+This is the only time when the generated "suite" file contains the useful 
+information. 
+
+The top-level `testsuite` tag is named `JUnit Platform Suite`, and contains 
+a `testcase` entry for each test run in any of the classes listed in the 
+suite. The `name` attribute for each of these `testcase` tags is the method 
+name of the test, with type information (eg. `example()`), but each has the 
+`classname` attribute set properly so it's possible to distinguish the test 
+cases from one another.
+
+#### Dynamic
+
+Generates a top-level `testsuite` named `JUnit Jupiter`, containing a 
+`testcase` for each generated test case. These are named after the 
+generating method, but with a number appended (eg. `factory()[1]`)
+
+#### Parameterised
+
+This is exactly the same as the output generated by Maven, with the twist 
+that the top-level `testsuite` is named `JUnit Jupiter`. The individual 
+`testcase` tag have a `name` attribute matching the automatically generated 
+test name (eg. `testMethod{String)[1]`).
+
+#### Pathological
+
+Creates a single test class with a top-level `testsuite` tag named `JUnit
+Jupiter`. This contains a single `testcase` tag with a `name` attribute 
+matching the method signature of the ignored test case (eg. `example(String)
+`) no matter how many tests would have been generated. The single `testcase` 
+contains a nested `skipped` tag, containing the reason for the test being 
+skipped.
+
+### Bazel's JUnit4 runner
+
+Obviously, not every test type supported by JUnit5 can be represented by 
+JUnit4, but for those that can....
+
+In all cases, Bazel generates a single file per test target. The generated 
+`testsuite` tags each have an empty `properties` tag. Bazel is also the only 
+test runner that generates a top-level `testsuites` tag for each file.
+
+#### Simple
+
+Below the `testsuites` tag is a single `testsuite` tag. This is named after 
+the test class name. Nested below this is a `testcase` for each test method. 
+The `name` attribute is the name of the test method without type information 
+(eg. `example`)
+
+#### Suite
+
+#### Dynamic
+
+Skipped
+
+#### Parameterised
+
+Parameterised tests work slightly differently in JUnit 4 than they do in 
+JUnit5. In JUnit4, each parameter is passed to the constructor of the test 
+class, meaning that each test class must store that value on a field for it 
+to be accessible to test methods.
+
+Below the `testsuites` tag is a `testsuite` tag for the test class itself. 
+This is basically empty.
+
+For each generated test case, there is a `testsuite` entry, named after the 
+order in which the suite was generated (eg. `[0]` or `[1]`). Within each of 
+those `testsuite` entries is a `testcase` per method in the test class, 
+where each `testcase` has a `name` attribute that is a combination of the 
+test method name without signature information and the run number (eg. 
+`someTest[1]`)
+
+#### Pathological
+
+Skipped
+
+## Desired behaviour
```

Comment:
Given all of the (wonderful!) effort that went into producing this spec, would it be possible to check in a golden XML file for the comparative tests, run the test, and assert that the XML matches? (even if we need to be a bit fuzzy zeroing-out timestamps and stuff)